protos/flyteidl/service/dataproxy.proto

```diff
@@ -63,6 +64,45 @@ message CreateDownloadLocationResponse {
   google.protobuf.Timestamp expires_at = 2;
 }
 
+// ArtifactType
+enum ArtifactType {
+  // ARTIFACT_TYPE_UNDEFINED is the default, often invalid, value for the enum.
+  ARTIFACT_TYPE_UNDEFINED = 0;
+
+  // ARTIFACT_TYPE_DECK refers to the deck html file optionally generated after a task, a workflow or a launch plan
+  // finishes executing.
+  ARTIFACT_TYPE_DECK = 1;
+}
+
+message CreateDownloadLinkRequest {
+  oneof source {
+    // WorkflowId is the unique identifier for the workflow execution.
+    core.WorkflowExecutionIdentifier execution_id = 1;
```

Comment:
I see in the implementation we only support node execution identifiers - why include workflow and task execution identifiers in the proto definition at all?